packages/plugin-core/src/test/suite-integ/TaskStatus.test.ts

```diff
@@ -0,0 +1,250 @@
+import _ from "lodash";
+import sinon from "sinon";
+import { expect, LocationTestUtils } from "../testUtilsv2";
+import { describeMultiWS, describeSingleWS } from "../testUtilsV3";
+import { before, after, describe } from "mocha";
+import { NoteProps, NoteUtils } from "@dendronhq/common-all";
+import { ExtensionProvider } from "../../ExtensionProvider";
+import { VSCodeUtils } from "../../vsCodeUtils";
+import { NoteTestUtilsV4, SinonStubbedFn } from "@dendronhq/common-test-utils";
+import { TaskStatusCommand } from "../../commands/TaskStatus";
+
+suite("GIVEN TaskStatus", function () {
+  describeSingleWS("WHEN a link to a task note is selected", {}, () => {
+    let taskNote: NoteProps;
+    let showQuickPick: SinonStubbedFn<typeof VSCodeUtils["showQuickPick"]>;
+    before(async () => {
+      const { engine, vaults, wsRoot } = ExtensionProvider.getDWorkspace();
+      const extension = ExtensionProvider.getExtension();
+      showQuickPick = sinon.stub(VSCodeUtils, "showQuickPick").resolves({
+        label: "y",
+      });
+      taskNote = await NoteTestUtilsV4.createNoteWithEngine({
+        engine,
+        fname: "task.test",
+        vault: vaults[0],
+        wsRoot,
+        body: "",
+        custom: {
+          status: "",
+        },
+      });
+      const currentNote = await NoteTestUtilsV4.createNoteWithEngine({
+        engine,
+        fname: "base",
+        vault: vaults[0],
+        wsRoot,
+        body: "[[task.test]]",
+      });
+      const editor = await ExtensionProvider.getWSUtils().openNote(currentNote);
+      editor.selection = LocationTestUtils.getPresetWikiLinkSelection();
+
+      const cmd = new TaskStatusCommand(extension);
+      await cmd.run();
+    });
+    after(() => {
+      showQuickPick.restore();
+    });
+
+    test("THEN prompts for the status", () => {
+      expect(showQuickPick.calledOnce).toBeTruthy();
+    });
+    test("THEN updates the task status", async () => {
+      const { engine } = ExtensionProvider.getDWorkspace();
+      const task = NoteUtils.getNoteByFnameFromEngine({
```

Comment:
nit: Use `engine.findNotes` instead

Reply:
Thanks, switched to it.

On a side note, could we add a "find 1 note" function for the engine too? It's just a little nicer in ergonomics, because otherwise you end up with `(await engine.find(...))[0]` when you are sure that there's only a single note.

Reply:
Sure, I can add a method on engine to make it a little simpler, Something like findFirstNote that returns a NoteProps or undefined.